### ISSUE TITLE
Reverse template to extract data from string already generated from a template

### ISSUE THREAD

**Issue by Rhvarrier** [pallets/jinja#1869]
I have a string that was already generated from a template using jinja. Later in another script I want to extract some information from this string using the same template. Could jinja provide a feature to match the formatted string against the template and then extract the information i.e do the inverse of what was done to generate the string. 

For example I have the following template

```This is a dummy file by {{ name }} containing :
    {{%- for data in data_list -%}}
    {{data.id}} {{data.value}}
    {{%- endfor -%}} ```

I generate the following string:

```This is a dummy file by Sam containing:
    1 foo
    2 bar```

Later I want to extract the dictionary using the same template
    ```{ name: "Sam",
    data_list: [{id: 1, value: "foo"}, {id: 2, value: "bar"}]}```

**Comment by ThiefMaster** [comment]
No. This sounds like an absolutely crazy amount of complexity (or something that's actually impossible) for almost no gain. Seems like something that should be handled on your application side, e.g. by dumping the template context into a JSON object...
